added placeholder files

baseballlover723 opened the pull request:
#1

Commit:
added base files

Commit:
added client and server mixin. renamed file to use snake case

Trottasn reviewed (dismissed):
All of this will totally work for my tcp implementation that I already have. I think I can pull this stuff in and put my code where it needs to be.

cbudo reviewed (dismissed):
All of this looks good! is there a reason that we're storing the tcp/udp client&server in here?

Commit:
4/create benchmark for control (#6)

I mean, this all looks good. I'll fit my code into where it needs to go shortly and we'll go from there. Good work!

Trottasn commented on spec/control_classes/tcp_server.rb:
```diff
@@ -0,0 +1,7 @@
+class TCPServer
```
Probably not going to work exactly (unless we use something other than socket), because the TCPServer object itself is call this and it would shadow that naming.

Trottasn reviewed (approved):
I'll change the TCPServer name myself - other than this, it looks fine!
